app/helpers/patient_helper.rb

```diff
@@ -28,6 +33,26 @@ def normalize_and_get_state_name(name)
     state_names[normalize_name(name)] || nil
   end
 
+  # I want to call this function in other files. Is it okay to use `self.` here so that I can do that?
+  # @twong @kfagan @tstrass @anyrubyexpert
+  def self.normalize_and_get_language_name(lang)
+    return nil if lang.nil?
+    return lang if lang == 'spa-PR' # 'spa-PR' is the only case-sensitive language code
+
+    lang = lang.to_s.downcase
+    # tries to match lang to either a 3-letter iso code or a language name
+    # If able to match, returns the 3-letter iso code for that language
+    # If unable to match, returns nil
+
+    # first search in all 3-letter language codes
+    matched_language = nil
+    matched_language = PATIENT_HELPER_FILES[:languages][lang.to_sym][:code] if PATIENT_HELPER_FILES[:languages][lang.to_sym]
```

Comment:
In general I'm wondering why we need `code`? If the key is already always the `code`, do we need to duplicate that information in `languages.yaml`? Or are there cases where `code` differs from the key? Or cases where it's just much easier to have `code` defined.

Reply:
Good point! I just confirmed that all `code`s were exact matches of the `key`s so, `:code` has been removed!